speed-dreams/source-2.2.3/src/modules/userinterface/legacymenu/mainscreens/ResearcherMenu.cpp

```diff
@@ -198,13 +198,18 @@ static void SetUserID(void*)
 /// @brief Saves the settings into the frontend settings and the backend config
 static void SaveSettings(void* /* dummy */)
 {
-    // Save settings to backend config
+    // Save settings to the SDAConfig
     SMediator* mediator = SMediator::GetInstance();
     mediator->SetTask(m_task);
     mediator->SetIndicatorSettings(m_indicators);
     mediator->SetInterventionType(m_interventionType);
     mediator->SetMaxTime(m_maxTime);
-    mediator->SetUserID(m_userID);
+
+    // Save the encrypted userID in the SDAConfig
+    char buf[32];
+    size_t userID_encrypted = std::hash<std::string>{}(m_userID);
+    sprintf(buf, "%d", userID_encrypted);
```

Comment:
Variable naming must be camelCase and dont capitalize whole abbreviations(ID)
```suggestion
    size_t encryptedUserId= std::hash<std::string>{}(m_userId);
    sprintf(buf, "%d", encryptedUserId);
```